# Obtención de estadísticas descriptivas

In [22]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns; sns.set_theme()

# Leer el archivo utilizando read_csv
data = pd.read_csv("covid19_tweets.csv")

In [3]:
# Verificar la cantidad de datos
data.count()

user_name           74436
user_location       59218
user_description    70079
user_created        74436
user_followers      74436
user_friends        74436
user_favourites     74436
user_verified       74436
date                74436
text                74436
hashtags            53002
source              74424
is_retweet          74436
dtype: int64

In [8]:
# Verificar las variables que contienen cada vector de datos y el tipo de variables
data.columns

Index(['user_name', 'user_location', 'user_description', 'user_created',
       'user_followers', 'user_friends', 'user_favourites', 'user_verified',
       'date', 'text', 'hashtags', 'source', 'is_retweet'],
      dtype='object')

In [5]:
data.dtypes

user_name           object
user_location       object
user_description    object
user_created        object
user_followers       int64
user_friends         int64
user_favourites      int64
user_verified         bool
date                object
text                object
hashtags            object
source              object
is_retweet            bool
dtype: object

Análisis de variables y rangos

In [12]:
data.iloc[:, 0].describe()

count                  74436
unique                 44853
top       GlobalPandemic.NET
freq                     312
Name: user_name, dtype: object

La variable user_name representa el nombre de usuario de la persona que realizó el tweet

In [13]:
data.iloc[:, 1].describe()

count     59218
unique    14622
top       India
freq       1496
Name: user_location, dtype: object

La variable user_location representa el lugar desde el que se realizó ese tweet, 
es posible observar que el lugar más común fue India

In [14]:
data.iloc[:, 2].describe()

count                                                 70079
unique                                                42690
top       Breaking News & Critical Information to SURVIV...
freq                                                    312
Name: user_description, dtype: object

La variable user_description representa la descripción del usuario que realizó el tweet. Podemos observar que el que más se repite tiene el mismo número de frecuencia que el user_name, esto es porque ese usuario fue el que más tweets sobre el covid 19 tiene, por lo tanto aparece que su username y su user_description es la más común. 

In [15]:
data.iloc[:, 3].describe()

count                   74436
unique                  45554
top       2010-07-13 21:58:05
freq                      312
Name: user_created, dtype: object

La variable user_created representa la fecha de creación de twitter del usuario que realizó el tweet

In [16]:
data.iloc[:, 4].describe()

count    7.443600e+04
mean     1.059513e+05
std      8.222900e+05
min      0.000000e+00
25%      1.660000e+02
50%      9.600000e+02
75%      5.148000e+03
max      1.389284e+07
Name: user_followers, dtype: float64

La variable user_followers representa la cantidad de usuarios que siguen a la cuenta que realizó el tweet, es posible observar que la mediana se encuentra en 10595, el mínimo número de seguidores es 0 y el máximo es 13892841 

In [17]:
data.iloc[:, 5].describe()

count     74436.000000
mean       2154.721170
std        9365.587474
min           0.000000
25%         153.000000
50%         552.000000
75%        1780.250000
max      497363.000000
Name: user_friends, dtype: float64

La variable user_friends representa la cantidad de personas que la cuenta que realizó el tweet sigue

In [18]:
data.iloc[:, 6].describe()

count    7.443600e+04
mean     1.529747e+04
std      4.668971e+04
min      0.000000e+00
25%      2.200000e+02
50%      1.927000e+03
75%      1.014800e+04
max      2.047197e+06
Name: user_favourites, dtype: float64

La varibale user_favourites representa la cantidad de personas en común (que sigue y lo siguen) de la cuenta que realizó el tweet

In [19]:
data.iloc[:, 7].describe()

count     74436
unique        2
top       False
freq      65082
Name: user_verified, dtype: object

La variable user_verified es un valor booleano que indica si la cuenta que realizó el tweet es una cuenta verificada, es importante notar que la mayor parte de las cuentas no están verificadas

In [20]:
data.iloc[:, 8].describe()

count                   74436
unique                  56546
top       2020-07-29 16:30:00
freq                       26
Name: date, dtype: object

La variable date indica la fecha en la cual se realizó un tweet, podemos observar que el día que hubieron más tweets sobre covid 19 fue el 29 de junio del 2020, con 26 tweets

In [21]:
data.iloc[:, 9].describe()

count                                                 74436
unique                                                74312
top       Greenland has no active cases of the novel cor...
freq                                                      6
Name: text, dtype: object

La variable text es donde se almacena el contenido del tweet.

In [22]:
data.iloc[:, 10].describe()

count           53002
unique          23445
top       ['COVID19']
freq            16004
Name: hashtags, dtype: object

La variable hashtags indica los hashtags que fueron agregados a un tweet, se puede observar que el hashtag más utilizado fue COVID19 con 16004 tweets

In [23]:
data.iloc[:, 11].describe()

count               74424
unique                450
top       Twitter Web App
freq                22974
Name: source, dtype: object

La varibale source representa el medio por el cual se realizó el tweet, se puede observar que la gran mayoría de los tweets se hicieron en la aplicación web de Twitter

In [24]:
data.iloc[:, 12].describe()

count     74436
unique        1
top       False
freq      74436
Name: is_retweet, dtype: object

Finalmente, la variable ir_retweet es un booleano que indica si el tweet fue retweet u original de la cuenta, podemos observar que la mayoría de los tweets son originales.

In [25]:
# Conclusiones a partir de los datos obtenidos
data.sort_values(by="user_followers", ascending=False).head(10)

user_name   user_location  \
6959       CGTN  Beijing, China   
13450      CGTN  Beijing, China   
16194      CGTN  Beijing, China   
235        CGTN  Beijing, China   
2837       CGTN  Beijing, China   
5344       CGTN  Beijing, China   
20483      CGTN  Beijing, China   
20378      CGTN  Beijing, China   
24243      CGTN  Beijing, China   
23721      CGTN  Beijing, China   

                                        user_description         user_created  \
6959   #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
13450  #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
16194  #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
235    #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
2837   #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
5344   #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
20483  #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
20378  #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
24243  #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   
23721  #SeeTheDifference with CGTN as we bring you st...  2013-01-24 03:18:59   

       user_followers  user_friends  user_favourites  user_verified  \
6959         13892841            69              104           True   
13450        13892839            69              104           True   
16194        13892837            69              104           True   
235          13892795            69              104           True   
2837         13892793            69              104           True   
5344         13892792            69              104           True   
20483        13892212            69              104           True   
20378        13892212            69              104           True   
24243        13892212            69              104           True   
23721        13892212            69              104           True   

                      date                                               text  \
6959   2020-07-25 08:00:00  #VoicesfromBeltandRoad: #COVID19 rap song aler...   
13450  2020-07-25 02:43:57  #China's civil aviation recovers as daily flig...   
16194  2020-07-25 00:27:38  On Friday, the #Chinese mainland reported:\n \...   
235    2020-07-25 12:20:00  #APEC reaffirms #COVID19 economic recovery pri...   
2837   2020-07-25 10:46:35  #COVID19 recovery can take weeks even for youn...   
5344   2020-07-25 09:03:28  #HongKong reports 133 new confirmed #COVID19 c...   
20483  2020-07-26 06:40:00  #COVID19 #HongKong SAR government introduces n...   
20378  2020-07-26 06:46:36  Global #COVID19 cases have surpassed 16 millio...   
24243  2020-07-26 02:31:57  Live: Students and recent grads from China and...   
23721  2020-07-26 03:04:23  #DPRK sees 1st suspected #COVID19 case, adopts...   

                                   hashtags                source  is_retweet  
6959   ['VoicesfromBeltandRoad', 'COVID19']  Twitter Media Studio       False  
13450                             ['China']       Twitter Web App       False  
16194               ['Chinese', 'Xinjiang']       Twitter Web App       False  
235                     ['APEC', 'COVID19']             TweetDeck       False  
2837                            ['COVID19']       Twitter Web App       False  
5344                ['HongKong', 'COVID19']       Twitter Web App       False  
20483               ['COVID19', 'HongKong']             TweetDeck       False  
20378                           ['COVID19']       Twitter Web App       False  
24243            ['coronavirus', 'COVID19']             Periscope       False  
23721                   ['DPRK', 'COVID19']       Twitter Web App       False

Podemos observar que la cuenta con más seguidores es CGTN, una cuenta verificada de Beijing, China que realizó diferentes tweets sobre el covid, incluso se puede ver la fluctuación de followers que tuvo a diferentes horas y en diferentes días

In [52]:
data.corr(method ='pearson')

user_followers  user_friends  user_favourites  user_verified  \
user_followers         1.000000     -0.002722        -0.028724       0.322896   
user_friends          -0.002722      1.000000         0.207825       0.013099   
user_favourites       -0.028724      0.207825         1.000000      -0.060316   
user_verified          0.322896      0.013099        -0.060316       1.000000   
is_retweet                  NaN           NaN              NaN            NaN   

                 is_retweet  
user_followers          NaN  
user_friends            NaN  
user_favourites         NaN  
user_verified           NaN  
is_retweet              NaN

Así mismo se puede observar que hay una correlación relativamente alta entre el número de seguidores y si una cuenta es verificada. Una cuenta verificada podría indicar que es una fuente confiable, sin embargo si lo que determina la verificación de la cuenta es el número de seguidores, no necesariamente se tratan de datos verídicos y comprobables.